src/gui/Shared/SpectrumAnalyser.h

```diff
@@ -18,11 +20,12 @@ class SpectrumAnalyser : public juce::Component,
     void setShowPostEQ (bool showPostSpectrum) { showPostEQ = showPostSpectrum; }
 
 private:
+    using optionalBackgroundTask = std::optional<std::reference_wrapper<gui::SpectrumAnalyserTask::SpectrumAnalyserBackgroundTask>>;
     juce::Path prePath;
     juce::Path postPath;
     const chowdsp::SpectrumPlotBase& eqPlot;
-    gui::SpectrumAnalyserTask::SpectrumAnalyserBackgroundTask& preTask;
-    gui::SpectrumAnalyserTask::SpectrumAnalyserBackgroundTask& postTask;
+    optionalBackgroundTask preTask;
+    optionalBackgroundTask postTask;
```

Comment:
Since we're using this alias in a number of places, maybe it would make sense to define the alias inside of `SpectrumAnalyserTask`. Like:
```cpp
class SpectrumAnalyserTask
{
public:
    using Optional = std::optional<std::reference_wrapper<SpectrumAnalyserBackgroundTask>>;
};
```
Then when we need to use it we can do `gui::SpectrumAnalyserTask::Optional`. Or something like that, I'm not sure I've got the names exactly right.